psamm/metabolicmodel.py

```diff
@@ -174,11 +196,11 @@ def database(self):
 
     @property
     def reactions(self):
-        return iter(self._reaction_set)
+        return self._reaction_set
```

Comment:
The returned value is not immutable. Although the `__setitem__` raises an exception the `append` method (and others) is still available so this allows the caller to modify the list of reactions without going through `add_reaction()` which would violate the class invariant.